src/core/ddsi/include/dds/ddsi/ddsi_xqos.h

```diff
@@ -228,11 +228,7 @@ typedef struct dds_type_consistency_enforcement_qospolicy {
   bool force_type_validation;
 } dds_type_consistency_enforcement_qospolicy_t;
 
-#ifdef DDS_HAS_SHM
-typedef struct dds_type_shared_memory {
-  unsigned char enabled;
-} dds_type_shared_memory_t;
-#endif
+typedef int dds_locator_mask_t;
```

Comment:
That should be an `uint32_t` — or, more precisely, an unsigned type of sufficient width, with 32 bits usually being a decent compromise. But mind the earlier remarks on using a bitmask to represent the set!

Reply:
You are correct on this, using signed entities for bitmasks can give very odd results, and explicitly stating the width is also a plus.
I will correct this.